File: frontend-service-provider-management-system/src/app/dto/service-provider-dto.ts
export class ServiceProviderDTO {
    id: number;
    serviceName: string;
    services: string[];
  
    constructor(id: number, serviceName: string, services: string[]) {
      this.id = id;
      this.serviceName = serviceName;
      this.services = services;
    }
  }
  

File: backend-service-provider-management-system/src/main/java/com/comping/serviceprovidermanagementsystem/dto/ServiceProviderDTO.java
package com.comping.serviceprovidermanagementsystem.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ServiceProviderDTO {
    private Long id;

    @NotEmpty(message = "serviceName field should not be null or empty")
    private String serviceName;

    @NotEmpty(message = "services field should not be null or empty")
    private List<String> services;

}
